src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java

```diff
@@ -213,6 +213,41 @@ public CqlSession getSession(String tenantId, String authToken, String userAgent
     return holder == null ? null : holder.session();
   }
 
+  /**
+   * Evicts a session from the cache based on the provided {@link RequestContext}.
+   *
+   * @see #evictSession(String, String, String) for details on eviction.
+   * @param requestContext The request context containing tenant, auth, and user agent info.
+   */
+  public void evictSession(RequestContext requestContext) {
+    Objects.requireNonNull(requestContext, "requestContext must not be null for eviction");
+
+    // Validation happens when creating the credentials and session key
```

Comment:
If this means missing values cannot happen, should a simple `get()` instead of `orElse()?